src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +115,38 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg)
```

Comment:
What happens if both `-a` and `--os` are specified? I'm not seeing how that case is handled. There should be a test for that.